### PR TITLE
refactor(konnect): add SDK interface types and theirs mocks

### PR DESCRIPTION
**Pull request by pmalek**
**What this PR does / why we need it**:

This PR will allow UTs for entity focused functions when mocks for those are introduced.

This PR does reorganize the `konnect` package a bit to solve the import cycles between `konnect` and newly added `ops` package.

Mocks are generated using [`mockery`](https://github.com/vektra/mockery) and `generate.mocks` makefile target. The config that configures which interfaces to generate mocks for is `.mockery.yaml`.

The tests that are added in this PR are in 
- `controller/konnect/ops/ops_controlplane_test.go`
- `controller/konnect/ops/ops_kongservice_test.go`

More will follow after this PR gets reviewed and merged.

**Which issue this PR fixes**

Part of #447

**Special notes for your reviewer**:

### CODE REVIEW

File: .golangci.yaml [hunk 1]
```diff
@@ -62,6 +62,13 @@ linters-settings:
 
       - pkg: github.com/kong/gateway-operator/internal/types
         alias: gwtypes
+
+      - pkg: github.com/Kong/sdk-konnect-go/models/components
+        alias: sdkkonnectgocomp
+      - pkg: github.com/Kong/sdk-konnect-go/models/operations
+        alias: sdkkonnectgoops
+      - pkg: github.com/Kong/sdk-konnect-go/models/sdkerrors
+        alias: sdkkonnectgoerrs
```
Comment (programmer04) [inline .golangci.yaml]: I think this `go` in the middle is redundant. WDYT e.g. `sdkkonnectops` instead of `sdkkonnectgoops`? The same for other packages

**Review by programmer04 — COMMENTED** [review]
Overall LGTM

File: .golangci.yaml (hunk 1 above)
Comment (randmonkey) [reply .golangci.yaml]: Changed the aliases.

File: .mockery.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+filename: "{{ trimSuffix .InterfaceFile \".go\" | base | lower }}_mock_test.go"
+dir: "{{ .InterfaceDir }}"
+mockname: "Mock{{ .InterfaceName }}"
+outpkg: "{{ .PackageName }}"
+
+packages:
+  github.com/kong/gateway-operator/controller/konnect/ops:
+    interfaces:
+      ControlPlaneSDK:
+      ServicesSDK:
```
Comment (czeslavo) [inline .mockery.yaml:15]: Nit: we might as well already generate mocks for other interfaces that are added in this PR (`ConsumerGroupSDK`, `ConsumersSDK`, `RoutesSDK`).

**Review by czeslavo — APPROVED** [review]

File: .mockery.yaml (hunk 2 above)
Comment (randmonkey) [reply .mockery.yaml:15]: I'll add mocks of all used SDKs in the following PR to refactor the usage of SDK.

**Pull request merged**